lib/doActionAwaitingNavigation.js

```diff
@@ -112,10 +112,10 @@ const waitForNavigation = (timeout, promises = []) => {
             );
           },
           defaultConfig.retryInterval,
-          defaultConfig.retryTimeout,
+          timeout,
```

Comment:
why should this be changed? `timeout` is used below to reject when timer expires.

Reply:
To honour the user given navigation timeout.. if default is used this can timeout before the user given time and the promise will get rejected.